packages/devtools_app/lib/src/error_badge_manager.dart

```diff
@@ -65,15 +102,43 @@ class ErrorBadgeManager extends DisposableController
     notifier.value = currentCount + 1;
   }
 
+  void appendInspectorErroredWidget(InspectableWidgetError error) {
+    var currentErrors = _activeErroredWidgets.value;
+    if (!currentErrors.any((e) => e.inspectorRef == error.inspectorRef)) {
+      currentErrors = [...currentErrors, error];
+      _activeErroredWidgets.value = currentErrors;
+      _errorCountNotifier(InspectorScreen.id).value = currentErrors.length;
+    }
+  }
+
   ValueListenable<int> errorCountNotifier(String screenId) {
     return _errorCountNotifier(screenId) ?? const FixedValueListenable<int>(0);
   }
 
+  ValueListenable<List<InspectableWidgetError>> erroredWidgetNotifier() {
```

Comment:
if you can implement my suggestion on `_activeErroredWidgets` above, this should take a string `screenId` and return the valueListenable for the screen

Reply:
Done! (6f779c11ea3b3f9617548fed1b7e3b4e8f51982b)